.gitignore

```diff
@@ -1,4 +1,252 @@
 
-public/
+# Created by https://www.toptal.com/developers/gitignore/api/macos,jetbrains+all,code,node,react
```

Comment:
Please don't put your personal choice of editor files in here.
See https://github.com/coderbyheart/first-principles/issues/30

Reply:
ok, I understand what you mean and I will revert it and handle it on my side, np 😄 

However, just as a healthy discussion, I don't completely agree because specially in open-source it's utopic to believe that everyone will have `a global git ignore file` that fits every use case and it's kind of an accident waiting to happen.

I wouldn't call it my personal preferences 😊 , It was generated using [gitignore.io](https://www.toptal.com/developers/gitignore) and I have been using it for so long that for me it feels kind of "standards", opinionated ones sure!

Thanks for the feedback will change it asap

Reply:
Done https://github.com/homeferences/website/pull/51/commits/17cc7756ded218edb94de82aea9a6d5f35d19b74